### PR TITLE
Cert-manager workaround for 2.1.0 and 2.1.1

### PR DESCRIPTION
**Pull request by gvasquezvargas**
## Jira Ticket

https://jira.d2iq.com/browse/D2IQ-87217

## Description of changes being made

This is a new PR that includes all suggestions and changes made in https://github.com/mesosphere/dcos-docs-site/pull/4402 but in the right places. Since these versions are live, we should be able to merge once eng gives the go.

The same workaround has to go in 2.1.2 but we will wait until everything is final here before we copy to https://github.com/mesosphere/dcos-docs-site/pull/4428.

### Preview

You can preview the docs build using the following URL, adding the PR # where specified:
http://docs-d2iq-com-pr-4478.s3-website-us-west-2.amazonaws.com/

### CODE REVIEW

**Review by palexster — APPROVED** [review]
/lgtm

File: pages/dkp/kommander/2.1/release-notes/2.1.0/index.md [hunk 1]
```diff
@@ -202,6 +202,22 @@ flux reconcile -n kommander-flux source git management --kubeconfig MANAGED_KUBE
...
+
+A permanent fix for the issue requires upgrading to Kommander 2.2.1 or higher. In the meantime, there is a workaround available that forces the applications to reconcile and recognize the renewed certificate. This workaround also extends the validity of the certificates to 10 years, fixes the certification reload issue, and restarts the affected pods once the new certificate is issued.
+
+The workaround applies to any environment (networked, air-gapped, on-prem, etc.) and fixes the issue regardless of your issuer type ([SelfSigned][selfsigned] for air-gapped environments, [ACME][acme], or your own certificate issuer configured separately for your institution).
```
Comment (cneth) [inline pages/dkp/kommander/2.1/release-notes/2.1.0/index.md:211]:
```suggestion
The docker container can be applied to the management cluster in any environment (networked, air-gapped, on-prem, etc.) and remediate the issue regardless of your CA issuer type ([SelfSigned][selfsigned] for air-gapped environments, [ACME][acme], or your own certificate issuer configured separately for your institution).   

If there are any workload clusters attached to the management cluster, they will also be remediated.

Note that this container does affect or change in any way certificates associated with a custom domain you may have created for the cluster.  
```

File: pages/dkp/kommander/2.1/release-notes/2.1.0/index.md [hunk 2]
```diff
@@ -202,6 +202,22 @@ flux reconcile -n kommander-flux source git management --kubeconfig MANAGED_KUBE
...
+
+To **prevent your applications from breaking**, or to **get the nodes up and running again**, and fix this issue permanently, run this command:
```
Comment (cneth) [inline pages/dkp/kommander/2.1/release-notes/2.1.0/index.md:213]: I am not sure the emphasis adds much here

```suggestion
To fix the issue on an impacted cluster,  run this command:
```

File: pages/dkp/kommander/2.1/release-notes/2.1.0/index.md [hunk 3]
```diff
@@ -202,6 +202,22 @@ flux reconcile -n kommander-flux source git management --kubeconfig MANAGED_KUBE
 ✔ fetched revision main/GIT_HASH
 ```
 
+### Cert-manager expiration workaround
+
+Due to an oversight, some Kommander versions do not properly handle certificate renewal for Kommander applications. The`cert-manager` component renews all certificates 60 days after you install Kommander on your cluster. When this occurs, some Kommander applications and pods fail to receive the renewed certificate information, causing them to stop working upon expiration. This occurs 90 days after Kommander was installed, which normally would coincide with the date you created the cluster. While the effects can vary, the most common failure is the inability to log in to the UI due to an expired certificate in the dex-k8s-authenticator pod.
```
Comment (cneth) [inline pages/dkp/kommander/2.1/release-notes/2.1.0/index.md:207]:
```suggestion
Due to an oversight, some Kommander versions do not properly handle certificate renewal for the Cluster CA and certificates that are created for Kommander applications. The`cert-manager` component renews all certificates 60 days after you install Kommander on your cluster. When this occurs, some Kommander applications and pods fail to receive the renewed certificate information, causing them to stop working upon expiration. This occurs 90 days after Kommander was installed, which normally would coincide with the date you created the cluster. While the effects can vary, the most common failure is the inability to log in to the UI due to an expired certificate in the dex-k8s-authenticator pod.  You may also see issues with flux no longer being able to access the internal gitea repository.
```

**Review by cneth — CHANGES_REQUESTED** [review]

File: pages/dkp/kommander/2.1/release-notes/2.1.1/index.md [hunk 4]
```diff
@@ -199,6 +199,22 @@ flux reconcile -n kommander-flux source git management --kubeconfig MANAGED_KUBE
 ✔ fetched revision main/GIT_HASH
 ```
 
+### Cert-manager expiration workaround
+
+Due to an oversight, some Kommander versions do not properly handle certificate renewal for Kommander applications. The`cert-manager` component renews all certificates 60 days after you install Kommander on your cluster. When this occurs, some Kommander applications and pods fail to receive the renewed certificate information, causing them to stop working upon expiration. This occurs 90 days after Kommander was installed, which normally would coincide with the date you created the cluster. While the effects can vary, the most common failure is the inability to log in to the UI due to an expired certificate in the dex-k8s-authenticator pod.
```
Comment (gvasquezvargas) [inline pages/dkp/kommander/2.1/release-notes/2.1.1/index.md:204]: Due to an oversight, some Kommander versions do not properly handle certificate renewal for the Cluster CA and certificates that are created for Kommander applications. The`cert-manager` component renews all certificates 60 days after you install Kommander on your cluster. When this occurs, some Kommander applications and pods fail to receive the renewed certificate information, causing them to stop working upon expiration. This occurs 90 days after Kommander was installed, which normally would coincide with the date you created the cluster. While the effects can vary, the most common failure is the inability to log in to the UI due to an expired certificate in the dex-k8s-authenticator pod. You may also see issues with flux no longer being able to access the internal gitea repository.

Comment (gvasquezvargas) [inline pages/dkp/kommander/2.1/release-notes/2.1.1/index.md:204]:
```suggestion
Due to an oversight, some Kommander versions do not properly handle certificate renewal for the Cluster CA and certificates that are created for Kommander applications. The`cert-manager` component renews all certificates 60 days after you install Kommander on your cluster. When this occurs, some Kommander applications and pods fail to receive the renewed certificate information, causing them to stop working upon expiration. This occurs 90 days after Kommander was installed, which normally would coincide with the date you created the cluster. While the effects can vary, the most common failure is the inability to log in to the UI due to an expired certificate in the dex-k8s-authenticator pod. You may also see issues with flux no longer being able to access the internal gitea repository.
```

File: pages/dkp/kommander/2.1/release-notes/2.1.1/index.md [hunk 5]
```diff
@@ -199,6 +199,22 @@ flux reconcile -n kommander-flux source git management --kubeconfig MANAGED_KUBE
...
+
+A permanent fix for the issue requires upgrading to Kommander 2.2.1 or higher. In the meantime, there is a workaround available that forces the applications to reconcile and recognize the renewed certificate. This workaround also extends the validity of the certificates to 10 years, fixes the certification reload issue, and restarts the affected pods once the new certificate is issued.
```
Comment (gvasquezvargas) [inline pages/dkp/kommander/2.1/release-notes/2.1.1/index.md:206]:
```suggestion
A permanent fix for the issue requires upgrading to Kommander 2.2.1 or higher. In the meantime, a docker container is available that contains a script that extends the validity of the Cluster CA to 10 years, fixes the certificate reload issue, and restarts the affected pods once the new certificates are issued.
```

File: pages/dkp/kommander/2.1/release-notes/2.1.0/index.md (hunk 1 above)
Comment (gvasquezvargas) [reply pages/dkp/kommander/2.1/release-notes/2.1.0/index.md:211]: @cneth just double-checking. I think there is a "not" missing in line 215? 

```suggestion
Note that this container does not affect or change in any way certificates associated with a custom domain you may have created for the cluster.  
```

File: pages/dkp/kommander/2.1/release-notes/2.1.1/index.md [hunk 6]
```diff
@@ -199,6 +199,22 @@ flux reconcile -n kommander-flux source git management --kubeconfig MANAGED_KUBE
...
+
+The workaround applies to any environment (networked, air-gapped, on-prem, etc.) and fixes the issue regardless of your issuer type ([SelfSigned][selfsigned] for air-gapped environments, [ACME][acme], or your own certificate issuer configured separately for your institution).
+
+To **prevent your applications from breaking**, or to **get the nodes up and running again**, and fix this issue permanently, run this command:
```
Comment (gvasquezvargas) [inline pages/dkp/kommander/2.1/release-notes/2.1.1/index.md:210]:
```suggestion
To fix the issue on an impacted cluster,  run this command:
```

File: pages/dkp/kommander/2.1/release-notes/2.1.0/index.md [hunk 7]
```diff
@@ -202,6 +202,22 @@ flux reconcile -n kommander-flux source git management --kubeconfig MANAGED_KUBE
 ✔ fetched revision main/GIT_HASH
 ```
 
+### Cert-manager expiration workaround
+
+Due to an oversight, some Kommander versions do not properly handle certificate renewal for the Cluster CA and certificates that are created for Kommander applications. The`cert-manager` component renews all certificates 60 days after you install Kommander on your cluster. When this occurs, some Kommander applications and pods fail to receive the renewed certificate information, causing them to stop working upon expiration. This occurs 90 days after Kommander was installed, which normally would coincide with the date you created the cluster. While the effects can vary, the most common failure is the inability to log in to the UI due to an expired certificate in the dex-k8s-authenticator pod. You may also see issues with flux no longer being able to access the internal gitea repository.
```
Comment (gvasquezvargas) [inline pages/dkp/kommander/2.1/release-notes/2.1.0/index.md:207]: Synced with Tarun and the clusters could start partially breaking after 60 days. 
```suggestion
Due to an oversight, some Kommander versions do not properly handle certificate renewal for the Cluster CA and certificates that are created for Kommander applications. The`cert-manager` component renews all certificates 60 days after you install Kommander on your cluster. When this occurs, some Kommander applications and pods fail to receive the renewed certificate information, causing them to stop working upon expiration. This occurs 60-90 days after Kommander was installed, which normally would coincide with the date you created the cluster. While the effects can vary, the most common failure is the inability to log in to the UI due to an expired certificate in the dex-k8s-authenticator pod. You may also see issues with flux no longer being able to access the internal gitea repository.
```

File: pages/dkp/kommander/2.1/release-notes/2.1.1/index.md [hunk 8]
```diff
@@ -199,6 +199,26 @@ flux reconcile -n kommander-flux source git management --kubeconfig MANAGED_KUBE
 ✔ fetched revision main/GIT_HASH
 ```
 
+### Cert-manager expiration workaround
+
+Due to an oversight, some Kommander versions do not properly handle certificate renewal for the Cluster CA and certificates that are created for Kommander applications. The`cert-manager` component renews all certificates 60 days after you install Kommander on your cluster. When this occurs, some Kommander applications and pods fail to receive the renewed certificate information, causing them to stop working upon expiration. This occurs 90 days after Kommander was installed, which normally would coincide with the date you created the cluster. While the effects can vary, the most common failure is the inability to log in to the UI due to an expired certificate in the dex-k8s-authenticator pod. You may also see issues with flux no longer being able to access the internal gitea repository.
```
Comment (gvasquezvargas) [inline pages/dkp/kommander/2.1/release-notes/2.1.1/index.md:204]:
```suggestion
Due to an oversight, some Kommander versions do not properly handle certificate renewal for the Cluster CA and certificates that are created for Kommander applications. The`cert-manager` component renews all certificates 60 days after you install Kommander on your cluster. When this occurs, some Kommander applications and pods fail to receive the renewed certificate information, causing them to stop working upon expiration. This occurs 60-90 days after Kommander was installed, which normally would coincide with the date you created the cluster. While the effects can vary, the most common failure is the inability to log in to the UI due to an expired certificate in the dex-k8s-authenticator pod. You may also see issues with flux no longer being able to access the internal gitea repository.
```

File: pages/dkp/kommander/2.1/release-notes/2.1.0/index.md (hunk 1 above)
Comment (msdolbey) [reply pages/dkp/kommander/2.1/release-notes/2.1.0/index.md:211]: Made an edit - Docker should be caps'ed

**Comment by msdolbey** [comment]
@gvasquezvargas - I have a hitchhiker commit for something that I noticed in a latter part of the notes section - there were some unclosed `<code>` tags that I just sorted out.

File: pages/dkp/kommander/2.1/release-notes/2.1.0/index.md [hunk 9]
```diff
@@ -202,6 +202,22 @@ flux reconcile -n kommander-flux source git management --kubeconfig MANAGED_KUBE
 ✔ fetched revision main/GIT_HASH
 ```
 
+### Cert-manager expiration workaround
```
Comment (msdolbey) [inline pages/dkp/kommander/2.1/release-notes/2.1.0/index.md:205]: Should this be lowercase?

```suggestion
### cert-manager expiration workaround
```

File: pages/dkp/kommander/2.1/release-notes/2.1.1/index.md [hunk 10]
```diff
@@ -199,6 +199,26 @@ flux reconcile -n kommander-flux source git management --kubeconfig MANAGED_KUBE
 ✔ fetched revision main/GIT_HASH
 ```
 
+### Cert-manager expiration workaround
```
Comment (msdolbey) [inline pages/dkp/kommander/2.1/release-notes/2.1.1/index.md:202]:
```suggestion
### cert-manager expiration workaround
```

File: pages/dkp/kommander/2.1/release-notes/2.1.0/index.md (hunk 7 above)
Comment (msdolbey) [reply pages/dkp/kommander/2.1/release-notes/2.1.0/index.md:207]: Updated spacing on the cert-manager above

File: pages/dkp/kommander/2.1/release-notes/2.1.1/index.md (hunk 8 above)
Comment (msdolbey) [reply pages/dkp/kommander/2.1/release-notes/2.1.1/index.md:204]: updating spacing on thecert-manager above

**Review by msdolbey — APPROVED** [review]
👍 
This looks good to me - I haven't gotten a chance to try it IRL yet, but I think Craig has done more of that than myself. Looks good to me.
Some small nits, but otherwise 👍 
nice workaround

**Review by cneth — COMMENTED** [review]
This is good, we just need to finish the testing, saw some last minute issues with attached clusters we're ironing out right now.

File: pages/dkp/kommander/2.1/release-notes/2.1.0/index.md [hunk 11]
```diff
@@ -202,6 +202,26 @@ flux reconcile -n kommander-flux source git management --kubeconfig MANAGED_KUBE
...
+
+A permanent fix for the issue requires upgrading to Kommander 2.2.1 or higher. In the meantime, a docker container is available that contains a script that extends the validity of the Cluster CA to 10 years, fixes the certificate reload issue, and restarts the affected pods once the new certificates are issued.
+
+The Docker container can be applied to the management cluster in any environment (networked, air-gapped, on-prem, etc.) and remediate the issue regardless of your CA issuer type ([SelfSigned][selfsigned] for air-gapped environments, [ACME][acme], or your own certificate issuer configured separately for your institution).
```
Comment (cneth) [inline pages/dkp/kommander/2.1/release-notes/2.1.0/index.md:211]:
```suggestion
The Docker container can be applied to the management cluster in any environment (networked, air-gapped, on-prem, etc.) and remediates the issue regardless of your CA issuer type ([SelfSigned][selfsigned] for air-gapped environments, [ACME][acme], or your own certificate issuer configured separately for your institution).
```

File: pages/dkp/kommander/2.1/release-notes/2.1.1/index.md [hunk 12]
```diff
@@ -199,6 +199,26 @@ flux reconcile -n kommander-flux source git management --kubeconfig MANAGED_KUBE
...
+
+A permanent fix for the issue requires upgrading to Kommander 2.2.1 or higher. In the meantime, a docker container is available that contains a script that extends the validity of the Cluster CA to 10 years, fixes the certificate reload issue, and restarts the affected pods once the new certificates are issued.
+
+The docker container can be applied to the management cluster in any environment (networked, air-gapped, on-prem, etc.) and remediate the issue regardless of your CA issuer type ([SelfSigned][selfsigned] for air-gapped environments, [ACME][acme], or your own certificate issuer configured separately for your institution).
```
Comment (cneth) [inline pages/dkp/kommander/2.1/release-notes/2.1.1/index.md:208]:
```suggestion
The docker container can be applied to the management cluster in any environment (networked, air-gapped, on-prem, etc.) and remediates the issue regardless of your CA issuer type ([SelfSigned][selfsigned] for air-gapped environments, [ACME][acme], or your own certificate issuer configured separately for your institution).
```

**Review by cmastr — APPROVED** [review]

**Pull request merged**